packages/cli/generators/relation/utils.generator.js

```diff
@@ -0,0 +1,235 @@
+const path = require('path');
+const tsquery = require('../../lib/ast-helper');
+const utils = require('../../lib/utils');
+
+exports.relationType = {
+  belongsTo: 'belongsTo',
+  hasMany: 'hasMany',
+};
+
+class AstLoopBackProject extends ast.Project {
+  constructor() {
+    super({
+      manipulationSettings: {
+        indentationText: ast.IndentationText.TwoSpaces,
+        insertSpaceAfterOpeningAndBeforeClosingNonemptyBraces: false,
+        newLineKind: ast.NewLineKind.LineFeed,
+        quoteKind: ast.QuoteKind.Single,
+      },
+    });
+  }
+}
+exports.AstLoopBackProject = AstLoopBackProject;
+
+exports.getModelPrimaryKeyProperty = async function(fs, modelDir, modelName) {
+  const modelFile = path.join(modelDir, utils.getModelFileName(modelName));
+
+  const fileContent = await fs.read(modelFile, {});
+  return tsquery.getIdFromModel(fileContent);
+};
+
+exports.getModelPropertyType = function(modelDir, modelName, propertyName) {
+  const project = new this.AstLoopBackProject();
+
+  const modelFile = path.join(modelDir, utils.getModelFileName(modelName));
+  const sf = project.addExistingSourceFile(modelFile);
+  const co = this.getClassObj(sf, modelName);
+  return this.getPropertyType(co, propertyName);
+};
+
+exports.addFileToProject = function(project, dir, modelName) {
+  const fileName = path.resolve(dir, utils.getModelFileName(modelName));
+  return project.addExistingSourceFile(fileName);
+};
+
+exports.getClassObj = function(fileName, modelName) {
+  return fileName.getClassOrThrow(modelName);
+};
+
+exports.getClassConstructor = function(classObj) {
+  return classObj.getConstructors()[0];
+};
+
+exports.addExportController = async function(
+  generator,
+  fileName,
+  controllerClassName,
+  controllerFileName,
+) {
+  const project = new this.AstLoopBackProject();
+  let pFile;
+  const exportDeclaration = {
+    moduleSpecifier: './' + controllerFileName,
+  };
+
+  if (generator.fs.exists(fileName)) {
+    pFile = project.addExistingSourceFile(fileName);
+    for (const declaration of pFile.getExportedDeclarations()) {
+      if (
+        ast.TypeGuards.isClassDeclaration(declaration) &&
+        controllerClassName === declaration.getName()
+      ) {
+        return;
+      }
+    }
+    pFile.addExportDeclaration(exportDeclaration);
+  } else {
+    pFile = project.createSourceFile(fileName, {
+      exports: [exportDeclaration],
+    });
+  }
+
+  await pFile.save();
+};
+
+/**
+ * Validate if property exist in class.
+ *
+ * @param {classObj}
+ * @param {propertyName} string
+ *
+ * @return bool true on success, false on failure.
+ */
+
+exports.doesPropertyExist = function(classObj, propertyName) {
+  return classObj
+    .getProperties()
+    .map(x => x.getName())
+    .includes(propertyName);
+};
+
+exports.doesRelationExist = function(classObj, propertyName) {
+  if (this.doesPropertyExist(classObj, propertyName)) {
+    throw new Error('property ' + propertyName + ' exists in the model');
```

Comment:
```suggestion
    throw new Error('property ' + propertyName + ' already exists in the model');
```

It would be great to include the model name in the message too, e.g.

```
property category already exists in the model Product
```

This scenario is not covered by any tests AFAICT, could you please add one?

Reply:
Fixed: updated message and added test.